File: Solutions/pe_008.py
# project euler-008

def largest_product(n,k): 
    product_list=[]
    for i in range(0,n-k+1):
        Product=1
        for j in range(i,i+k):
            Product*=int(D[j])
        product_list.append(Product)
    return max(product_list)
    
t=int(input())

for i in range(t):
    a=[int(x) for x in input().split()]
    N=a[0]
    K=a[1]
    D=str(input())
    print(largest_product(N,K))

File: Clojure/pe-008.clj
(defn product-number-str [some-str]
  (->> (seq some-str)
       (map str)
       (map #(Integer/parseInt %))
       (reduce *)))

(defn thirteen-subs-largest-product [some-str]
  (->> (range 0 988)
       (map #(subs some-str % (+ 13 %)))   
       (map product-number-str)
       (apply max)))

(def input-str (->> (take-while #(not= "" %) (repeatedly #(.readLine *in*)))
                    (reduce str)))

;Press Enter twice after you input the 1000-number string at stdin

(println (thirteen-subs-largest-product input-str))
